dashboard/client/src/common/MultiTabLogViewer.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Box,
+  createStyles,
+  IconButton,
+  makeStyles,
+  Tab,
+  Tabs,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import { RiArrowUpDownLine } from "react-icons/ri";
+import { useStateApiLogs } from "../pages/log/hooks";
+import { LogViewer } from "../pages/log/LogViewer";
+import { HideableBlock } from "./CollapsibleSection";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    tabs: {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+    },
+  }),
+);
+
+export type MultiTabLogViewerTab = {
+  title: string;
+  nodeId: string | null;
+  filename?: string;
+};
+
+export type MultiTabLogViewerProps = {
+  tabs: MultiTabLogViewerTab[];
+};
+
+export const MultiTabLogViewer = ({ tabs }: MultiTabLogViewerProps) => {
+  const classes = useStyles();
+  const [value, setValue] = useState(tabs[0]?.title);
+  const [expanded, setExpanded] = useState(false);
+
+  const currentTab = tabs.find((tab) => tab.title === value);
+
+  return (
+    <div>
+      <Box
+        display="flex"
+        flexDirection="row"
+        alignItems="flex-start"
+        justifyContent="space-between"
+      >
+        <Box
+          display="flex"
+          flexDirection="column"
+          alignItems="stretch"
+          flexGrow={1}
+        >
+          <Tabs
+            className={classes.tabs}
+            value={value}
+            onChange={(_, newValue) => {
+              setValue(newValue);
+            }}
+          >
+            {tabs.map(({ title }) => (
+              <Tab key={title} label={title} value={title} />
+            ))}
+          </Tabs>
+
+          {!currentTab ? (
+            <Typography color="error">Please select a tab.</Typography>
+          ) : (
+            tabs.map(({ title, nodeId, filename }) => (
+              <HideableBlock
+                key={title}
+                visible={title === currentTab?.title}
+                keepRendered
+              >
+                <StateApiLogViewer
+                  nodeId={nodeId}
+                  filename={filename}
+                  height={expanded ? 800 : 300}
```

Comment:
nit: extract as consts. Seems like 300 is also set as default in `StateApiLogViewer`